### PR TITLE
Adding IP address obfuscation

### PR DESCRIPTION
**Pull request by epacke**
I've added a menu at the top left of the page. This is not as much thought through as the initial CodeQual app but I hope that it's fairly readable?

### CODE REVIEW

**Comment by haughki** [comment]
First questions:
image

**Comment by haughki** [comment]
This GitHub codereview interface is truly awful. I have no idea how to use it.
Regarding my previous comment with screenshot (I can't reply to that comment):
Ok, I sortof get it now. What's your plan for the 'dummy' text (the 192.168.10.1, etc.)? Are you going to pull that out, or is it meant to be some kind of "tutorial" data for the user?

File: code-quality/indeni.html [hunk 1]
```diff
@@ -4,20 +4,24 @@
 		<script src="./tippy.js"></script>
         <script src="./common.js"></script>
         <script src="./code-validation.js"></script>
+        <script src="./obfuscation.js"></script>
         <script src="./indeni-js-support.js"></script>
 		<script src="./indeni.js"></script>
         <link href="./indeni.css" rel="stylesheet" type="text/css">
 	</head>
 </html>
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/indeni.html]: Fixed.

File: code-quality/obfuscation.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $("pre#obfuscation-result-content").html($("textarea#input-textarea").val());
+    });
+    $("textarea#input-textarea").trigger("keyup");
+    
+    // Build the obfuscation options from the obfuscate functions
+    for(var name in obfuscateFunctions){
+
+        var f = obfuscateFunctions[name];
+        //Append the functions input data
+        $("div#obfuscation-types").append("<div id=\"" + name + "\">" + f.obfuscationOptionData + "</div>")
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/obfuscation.js]: Fixed.

File: code-quality/obfuscation.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+}
+
+var obfuscateFunctions = {
+
+    "ipAddresses": new function(){
+
+        // Obfuscates IP adresses
+        this.obfuscationOptionData = "Replaces IP addresses with the following /24 prefix (ie. 192.168.1): <input></input>";
+        this.obfuscationType = "input"
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/obfuscation.js]: Fixed

File: code-quality/obfuscation.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+
+function initiateObfuscations(){
+
+    // On each update the result field should be reset
+    $("textarea#input-textarea").on("keyup", function(){
+        $("pre#obfuscation-result-content").html($("textarea#input-textarea").val());
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/obfuscation.js]: Fixed

File: code-quality/indeni.html [hunk 5]
```diff
@@ -30,11 +34,28 @@
...
+        </div>
+        <div class="wrapper" id="obfuscation-wrapper">
+            <div class="header">
+                Paste your input here
+            </div>
+            <div class="header">
+                See the obfuscated result below
+            </div>
+            <div class="input-div">
+                <textarea id="input-textarea" cols=100 rows=20></textarea>
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/indeni.html]: Fixed

File: code-quality/obfuscation.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+            // Validate the input data
+            if(subnet.match(/^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/)){
+
+                // Show that the value is accepted by changing the background color to green
+                $(this).css("background", "#44EC88");
+                $(this).css("color", "black");
+
+                // Replace the subnets in the input file with the input value
+                content = content.replace(/[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/g, "<span class=\"obfuscated\">" + subnet + "</span>");
```
[the remark this answers is not in the document]
Comment (epacke) [reply code-quality/obfuscation.js]: Thank you for your patience. I'm still a newbie to these things.

Looked at the other code and right now I have one function that fetches the code, finds thing to obfuscate and alters the code to do the obfuscation.

Would it work better if:
run would get the code from the textarea and the pass it to another function which only takes code as a parameter and then returns it obfuscated? Should make it easy to make tests with that, right?

**Comment by epacke** [comment]
Never happened.

**Pull request closed without merging**